src/main/java/com/thebois/models/world/World.java

```diff
@@ -108,6 +106,19 @@ private void replaceTileInCanonicalMatrix(final ITile tile) {
         return emptyPositions;
     }
 
+    @Override
+    public Collection<Position> tryGetEmptyPositionsNextTo(
+        final Position position, final int maxCount, final float radius) {
+        return MatrixUtils.toCollection(canonicalMatrix).stream().filter(iTile -> !iTile
+            .getPosition()
+            .equals(position)
+                && iTile.getCost() < Float.MAX_VALUE
+                && radius > position.distanceTo(
+            iTile.getPosition()))
```

Comment:
This should be broken out into a boolean-returning method with a clear name describing what all these conditions mean.